idea/test-balance/src/services/gear.ts

```diff
@@ -0,0 +1,165 @@
+}
+
+export class GearService {
+  private account: KeyringPair;
+  private balanceToTransfer: BN;
+  private api: GearApi;
+  private genesis: string;
+  private providerAddress: string;
+  private queue: Array<TBRequestParams>;
+
+  constructor() {
+    this.providerAddress = config.gear.providerAddresses[0];
+    this.queue = [];
+  }
+
+  async init() {
+    this.account = await createAccount(config.gear.accountSeed);
+    this.balanceToTransfer = new BN(config.gear.balanceToTransfer);
+    await this.connect();
+    this.processQueue();
+  }
+
+  get genesisHash() {
+    return this.genesis;
+  }
+
+  async connect() {
+    if (!this.providerAddress) {
+      logger.error('There are no node addresses to connect to');
+      process.exit(1);
+    }
+
+    this.api = new GearApi({ providerAddress: this.providerAddress });
+
+    try {
+      await this.api.isReadyOrError;
+    } catch (error) {
+      logger.error(`Failed to connect to ${this.providerAddress}`, { error: error.message });
+      await this.reconnect();
+    }
+    await this.api.isReady;
+    this.api.on('disconnected', () => {
+      logger.error(`Disconnected from ${this.providerAddress}`);
+      this.reconnect();
+    });
+    this.genesis = this.api.genesisHash.toHex();
+    logger.info(`Connected to ${await this.api.chain()} with genesis ${this.genesis}`);
+    changeStatus('ws');
+  }
+
+  async reconnect(): Promise<void> {
+    this.genesis = null;
+    if (this.api) {
+      await this.api.disconnect();
+      this.api = null;
+    }
+
+    reconnectionsCounter++;
+    if (reconnectionsCounter > MAX_RECONNECTIONS) {
+      this.providerAddress = config.gear.providerAddresses.filter((address) => address !== this.providerAddress)[0];
+      reconnectionsCounter = 0;
+    }
+
+    logger.info('Attempting to reconnect');
+    changeStatus('ws');
+    return this.connect();
+  }
+
+  async sendBatch(addresses: string[]): Promise<[string[], string]> {
+    const txs = addresses.map((address) => this.api.balance.transfer(address, this.balanceToTransfer));
+    const batch = this.api.tx.utility.forceBatch(txs);
+    const transferred = [];
+    let blockHash: string;
+
+    const correlationId = randomUUID({});
+
+    logger.info(`Sending batch with ${addresses.length} transfers`, { addresses, correlationId });
+
+    await new Promise((resolve) => {
+      batch
+        .signAndSend(this.account, ({ events, status }) => {
+          if (status.isInBlock) {
+            blockHash = status.asInBlock.toHex();
+            logger.info(`Batch is in block`, { blockHash, correlationId });
+
+            for (const { event } of events) {
+              switch (event.method) {
+                case 'Transfer':
+                  transferred.push((event.data as TransferData).to.toHex());
+                  break;
+                case 'ExtrinsicSuccess':
+                  logger.info(`ExtrinsicSuccess`, { blockHash, correlationId });
+                  resolve(0);
```

Comment:
`0/1` can be moved to const with status names to better understanding

Reply:
refactored this part